--- origsrc/luasql-2.1.1/Makefile	2007-10-29 16:44:54.000000000 -0500
+++ src/luasql-2.1.1/Makefile	2008-02-13 18:58:07.031250000 -0600
@@ -24,7 +24,7 @@
 lib: src/$(LIBNAME)
 
 src/$(LIBNAME): $(OBJS)
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) -o $@ $(LIB_OPTION) $(OBJS) $(DRIVER_LIBS)
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(CFLAGS) -o $@ $(LIB_OPTION) $(OBJS) $(DRIVER_LIBS) $(LUA_LIBS)
 
 $(COMPAT_DIR)/compat-5.1.o: $(COMPAT_DIR)/compat-5.1.c
 	$(CC) -c $(CFLAGS) -o $@ $(COMPAT_DIR)/compat-5.1.c
--- origsrc/luasql-2.1.1/config	2007-10-27 17:55:27.000000000 -0500
+++ src/luasql-2.1.1/config	2008-02-13 18:59:46.953125000 -0600
@@ -24,10 +24,10 @@
 LUA_VERSION_NUM= 501
 
 # OS dependent
-LIB_OPTION= -shared #for Linux
+LIB_OPTION= -shared -Wl,--enable-auto-image-base
 #LIB_OPTION= -bundle -undefined dynamic_lookup #for MacOS X
 
-LIBNAME= $T.so
+LIBNAME= $T.dll
 COMPAT_DIR= ../compat/src
 
 # Compilation parameters
--- origsrc/luasql-2.1.1/src/ls_odbc.c	2007-08-22 13:37:06.000000000 -0500
+++ src/luasql-2.1.1/src/ls_odbc.c	2008-02-13 19:03:16.750000000 -0600
@@ -12,7 +12,7 @@
 #include <string.h>
 #include <time.h>
 
-#if defined(_WIN32)
+#if defined(_WIN32) || defined(__CYGWIN__)
 #include <windows.h>
 #include <sqlext.h>
 #elif defined(INFORMIX)
